presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/LongDecimalWithOverflowStateSerializer.java

```diff
@@ -33,21 +34,33 @@ public Type getSerializedType()
     }
 
     @Override
-    public void serialize(BigIntegerState state, BlockBuilder out)
+    public void serialize(LongDecimalWithOverflowState state, BlockBuilder out)
     {
-        if (state.getBigInteger() == null) {
+        if (state.getLongDecimal() == null) {
             out.appendNull();
         }
         else {
-            VARBINARY.writeSlice(out, Slices.wrappedBuffer(state.getBigInteger().toByteArray()));
+            byte[] longDecimalBytes = state.getLongDecimal().getBytes();
+
+            Slice slice = Slices.allocate(Long.BYTES + longDecimalBytes.length);
+            SliceOutput output = slice.getOutput();
+
+            output.writeLong(state.getOverflow());
+            output.writeBytes(longDecimalBytes);
+
+            VARBINARY.writeSlice(out, slice);
         }
     }
 
     @Override
-    public void deserialize(Block block, int index, BigIntegerState state)
+    public void deserialize(Block block, int index, LongDecimalWithOverflowState state)
     {
         if (!block.isNull(index)) {
-            state.setBigInteger(new BigInteger(VARBINARY.getSlice(block, index).getBytes()));
+            SliceInput slice = VARBINARY.getSlice(block, index).getInput();
+
+            state.setOverflow(slice.readLong());
+            byte[] longDecimalBytes = slice.readSlice(slice.available()).getBytes();
```

Comment:
why do you need `byte[]` intermediate array. I think you can directly put read slice in state